src/Orchard.Web/Modules/Orchard.MediaLibrary/Services/MediaLibraryService.cs

```diff
@@ -328,15 +328,16 @@ public class MediaLibraryService : IMediaLibraryService {
             Argument.ThrowIfNullOrEmpty(newFolderName, "newFolderName");
 
             try {
-                var segments = folderPath.Split(new char[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries).ToArray();
-                var newFolderPath = String.Join(Path.DirectorySeparatorChar.ToString(), segments.Take(segments.Length - 1).Union(new[] { newFolderName }));
+                var parentIndex = folderPath.LastIndexOfAny(new char[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar });
```

Comment:
No, String.Join with Path.DirectorySeparatorChar replaces "/" with "\\" in folder path. It is the primary cause of the issue.